src/Veldrid.Tests/TextureTests.cs

```diff
@@ -135,6 +135,568 @@ public unsafe void Update_ThenCopySingleMip_Succeeds_R16UNorm()
             }
         }
 
+
+        [Fact]
+        public void CreateTextureViewFromTextureWithArrayLayers()
+        {
+            const uint TexSize = 4;
+            const uint MipLevels = 1;
+            const uint ArrayLayers = 6;
+
+            TextureDescription texDesc = TextureDescription.Texture2D(
+                TexSize, TexSize, MipLevels, ArrayLayers, PixelFormat.R8_UNorm, TextureUsage.Storage | TextureUsage.Sampled);
+            Texture tex = RF.CreateTexture(texDesc);
+
+            for (uint mip = 0; mip < MipLevels; mip++)
+            {
+                for (uint layer = 0; layer < ArrayLayers; layer++)
+                {
+                    var mipSize = TexSize >> (int)mip;
+                    byte[] data = Enumerable.Repeat((layer + 1) * 42, (int)(mipSize * mipSize)).Select(n => (byte)n).ToArray();
+                    GD.UpdateTexture(tex, data, 0, 0, 0, mipSize, mipSize, 1, mip, layer);
+                }
+            }
+
+            var textureView = RF.CreateTextureView(tex);
+            Assert.NotNull(textureView);
+        }
+
+        [Fact]
+        public void CubeMap_UpdateAndRead()
+        {
+            const uint TexSize = 4;
+            const uint MipLevels = 3;
+
+            TextureDescription texDesc = TextureDescription.Texture2D(
+                TexSize, TexSize, MipLevels, 1, PixelFormat.R8_UNorm, TextureUsage.Cubemap);
+            Texture tex = RF.CreateTexture(texDesc);
+
+            for (uint mip = 0; mip < MipLevels; mip++)
+            {
+                for (uint face = 0; face < 6; face++)
+                {
+                    var mipSize = TexSize >> (int)mip;
+                    byte[] data = Enumerable.Repeat((face + 1) * 42, (int)(mipSize * mipSize)).Select(n => (byte)n).ToArray();
+                    GD.UpdateTexture(tex, data, 0, 0, 0, mipSize, mipSize, 1, mip, face);
+                }
+            }
+
+            Texture readback = GetReadback(tex);
+            Assert.All
```

Comment:
Though I found this to be the most pleasant assertion format to validate the data blocks in these tests, I still feel there must be a better way yet.

What I like about it is that it gives a better overview of the shape of the failure than simple loops because it aggregates the results instead of exiting after a single assertion failure.